### ISSUE TITLE
Source maps cause console spam when run with webpack

### ISSUE THREAD

**Issue by alschmiedt** [google/blockly#5615]
**Describe the bug**
The sources list in the [source maps file](https://github.com/google/blockly/blob/develop/blockly_compressed.js.map) we release contain references to nonexistent files (ex: synthetic:es6/util/createtemplatetagfirstarg). These were introduced with the recent move to goog.module. When we use our source maps in samples we run into problems where webpack emits a pile of warnings saying it can not find these files. We can remove that warning in samples, however this does not fix the problem for external developers.

**To Reproduce**

1. Go to field-date in blockly samples.
2. Install the latest beta version of Blockly. You might need to use `npm install --force`.
3. `npm run start`
4. Observe all the errors in terminal**

**Expected behavior**
We should have no warning messages in console. This bug is not to fix the problems happening in samples, but instead to research the problem and see if there is a fix that would also stop this from happening to developers using blockly + webpack.

**Output**
```
./node_modules/blockly/blockly_compressed.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/Users/aschmiedt/workspace/blockly-samples/plugins/field-date/node_modules/blockly/ [synthetic:base] ' file: Error: ENOENT: no such file or directory, open '/Users/aschmiedt/workspace/blockly-samples/plugins/field-date/node_modules/blockly/ [synthetic:base] '
```

**Additional context**
AFAICT source maps are still working. It just causes a lot of unnecessary output.

**Comment by maribethb** [comment]
These are caused by the Closure compiler. They are added by Closure for modules that it polyfills. I'm pretty sure that this issue: https://github.com/google/closure-compiler/issues/256 is related and it's been open since 2014 so I don't think they will fix it.

I did a quick test and just removing all the modules that start with "synthetic" (they are all at the beginning of the file) plus the one for "closure/goog/base_minimal.js" fixed the errors and didn't introduce new ones. The source maps still work and I was able to see the mapped sources for blockly inside of a plugin.

I think it should be fairly easy to write a script that removes these entries from the generated source map. I'm not sure when exactly it should be run, I guess whenever we generate new source maps (not sure exactly when that happens).

In the meantime, for anyone else that may be affected by this, you can see how we fixed this in our webpack config in https://github.com/google/blockly-samples/pull/982 (just silences the errors)

**Comment by cpcallen** [comment]
> I think it should be fairly easy to write a script that removes these entries from the generated source map. I'm not sure when exactly it should be run, I guess whenever we generate new source maps (not sure exactly when that happens).

I am insufficiently familiar with sourcemaps to have an opinion about whether this is the best solution, but I do think ideally such a script would go in [the gulp pipeline set up by buildCompiled, near the `gulp.sourcemaps.write` step](https://github.com/google/blockly/blob/00e1dade3ab5344f50156a60f95db7299884ae9b/scripts/gulpfiles/build_tasks.js#L594):

* Maybe before it, modifying the internal representation of the sourcemap directly (e.g. as we had been doing with `unflattenCorePaths`), or
* Maybe after it, modifying the sourcemap files which (if I read the code correctly) have been written to the vinyl stream but not yet to disk.

(I suspect the former is probably easier, but I've only glanced at [the implementation](https://github.com/gulp-sourcemaps/gulp-sourcemaps/).

**Comment by maribethb** [comment]
Okay so I tried using the gulp-sourcemaps api to remove the synthetic sources, and the closest I could come was to replace them with empty strings. Which to be fair seemed to work ok and not cause errors.

However something seems to have changed before this in the code because the source maps look different now. Run `npm run build:compiled` on a fresh develop and all the `[synthetic]` modules now look like `"src/com/google/javascript/jscomp/js/base.js"` instead. As far as I can tell, those files still don't exist either, but they don't utterly break webpack. Previous behavior was that when you run `npm run start` on a plugin you got 

```
Failed to parse source map from '/Users/maribethb/github/maribethb/blockly-samples/plugins/dev-tools/node_modules/blockly/ [synthetic:es6/util/makeiterator] ' file: Error: ENOENT: no such file or directory, open '/Users/maribethb/github/maribethb/blockly-samples/plugins/dev-tools/node_modules/blockly/ [synthetic:es6/util/makeiterator] '
```

and now you don't get any messages. However in dev tools if you actually try to go to that source file location when you try to open that file it says `Could not load content for webpack://@blockly/dev-tools/src/com/google/javascript/jscomp/js/es6/set.js?fa11 (HTTP error: status code 404, net::ERR_UNKNOWN_URL_SCHEME)`.

That seems fine since webpack isn't actually surfacing those errors. So I don't know what caused the synthetic files to be reformatted this way, I assume something else that y'all have been doing with compilation, but... seems like this might have accidentally been fixed.


In case that turns out not to be the case, for future reference I replaced the bad entries with empty strings by adding

```
.pipe(gulp.sourcemaps.mapSources((sourcePath, file) => {
        if (sourcePath.startsWith('src/com/google')) {
          return '';
        }
        return sourcePath;
      }))
```

to the build pipeline. There's this package that seemed like it might be relevant but I couldn't get it to work: https://github.com/gulp-sourcemaps/sources-content
